externals/kyuubi-flink-sql-engine/src/main/java/org/apache/kyuubi/engine/flink/context/ExecutionContext.java

```diff
@@ -0,0 +1,330 @@
+  }
+
+  /** Returns a builder for this {@link ExecutionContext}. */
+  public static Builder builder(
+      EngineEnvironment defaultEnv,
+      EngineEnvironment sessionEnv,
+      List<URL> dependencies,
+      Configuration configuration,
+      ClusterClientServiceLoader serviceLoader,
+      Options commandLineOptions,
+      List<CustomCommandLine> commandLines) {
+    return new Builder(
+        defaultEnv,
+        sessionEnv,
+        dependencies,
+        configuration,
+        serviceLoader,
+        commandLineOptions,
+        commandLines);
+  }
+
+  // ------------------------------------------------------------------------------------------------------------------
+  // Non-public methods
+  // ------------------------------------------------------------------------------------------------------------------
+
+  private TableEnvironmentInternal createStreamTableEnvironment(
+      StreamExecutionEnvironment env,
+      EnvironmentSettings settings,
+      TableConfig config,
+      Executor executor,
+      CatalogManager catalogManager,
+      ModuleManager moduleManager,
+      FunctionCatalog functionCatalog) {
+    final Map<String, String> plannerProperties = settings.toPlannerProperties();
+    final Planner planner =
+        ComponentFactoryService.find(PlannerFactory.class, plannerProperties)
+            .create(plannerProperties, executor, config, functionCatalog, catalogManager);
+
+    return new StreamTableEnvironmentImpl(
+        catalogManager,
+        moduleManager,
+        functionCatalog,
+        config,
+        env,
+        planner,
+        executor,
+        settings.isStreamingMode(),
+        classLoader);
+  }
+
+  private static Executor lookupExecutor(
+      Map<String, String> executorProperties, StreamExecutionEnvironment executionEnvironment) {
+    try {
+      ExecutorFactory executorFactory =
+          ComponentFactoryService.find(ExecutorFactory.class, executorProperties);
+      Method createMethod =
+          executorFactory
+              .getClass()
+              .getMethod("create", Map.class, StreamExecutionEnvironment.class);
+
+      return (Executor)
+          createMethod.invoke(executorFactory, executorProperties, executionEnvironment);
+    } catch (Exception e) {
+      throw new TableException(
+          "Could not instantiate the executor. Make sure a planner module is on the classpath", e);
+    }
+  }
+
+  private void initializeTableEnvironment() {
+    final EnvironmentSettings settings = engineEnvironment.getExecution().getEnvironmentSettings();
+    final TableConfig config = new TableConfig();
+    final ModuleManager moduleManager = new ModuleManager();
+    final CatalogManager catalogManager =
+        CatalogManager.newBuilder()
+            .classLoader(classLoader)
+            .config(config.getConfiguration())
+            .defaultCatalog(
+                settings.getBuiltInCatalogName(),
+                new GenericInMemoryCatalog(
```

Comment:
In spark engine, if we detect hive classes are located at classpath, then we use HiveCatalog as default, otherwise use memory catalog. The current implement LGTM now, we can discuss to unify the behavior later.